### PR TITLE
Can now remove entities with a MeshPaintComponent

### PR DESCRIPTION
**Pull request by jonathan2222**
### Additions
- Can now remove entities with extensions.

- Added a way to remove entities by pressing the **_delete_** key.

### Bug fixes
- Fixed validation error and memory leaks when removing entities. Somehow it did not release buffers when deleting the entity. This is now fixed for the most part. One problem which is remaining is a **_device lost_** error when enabling raytracing and trying to remove entities. This will occur after you have removed a few entities.

### CODE REVIEW

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 1]
```diff
@@ -1134,6 +1131,27 @@ namespace LambdaEngine
 			m_TLASDirty = true;
 		}
 
+		// Remove extension
+		{
+			// Fetch the current instance and its extension index.
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp:1136]: Unnecessary comment, the code speaks for itself here

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 2]
```diff
@@ -1134,6 +1131,27 @@ namespace LambdaEngine
...
+			Instance& currentInstance = rasterInstances[instanceIndex];
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: Can be a `const` reference. Same with `instanceKey` below.

File: LambdaEngine/Source/Rendering/EntityMaskManager.cpp [hunk 3]
```diff
@@ -18,6 +18,12 @@ bool LambdaEngine::EntityMaskManager::Init()
 	return true;
 }
 
+void LambdaEngine::EntityMaskManager::RemoveAllExtensionsFromEntity(Entity entity)
+{
+	if(s_EntityToExtensionGroupEntryMap.contains(entity))
+		s_EntityToExtensionGroupEntryMap.erase(entity);
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Rendering/EntityMaskManager.cpp]: Just `erase(entity)` is fine, it can handle misses.

**Review by TheoBerlin — CHANGES_REQUESTED** [review]

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 1 above)
Comment (jonathan2222) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp:1136]: Not true, I think this is better. Can see more clearly what the whole code block does.

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp (hunk 2 above)
Comment (jonathan2222) [reply LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]: 👍

File: LambdaEngine/Source/Rendering/EntityMaskManager.cpp (hunk 3 above)
Comment (jonathan2222) [reply LambdaEngine/Source/Rendering/EntityMaskManager.cpp]: 👌

**Review by TheoBerlin — APPROVED** [review]

**Pull request merged**